Meter.c

```diff
@@ -124,26 +125,48 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
    // First draw in the bar[] buffer...
    int offset = 0;
+   double actualBarWidth=0;
    for (uint8_t i = 0; i < this->curItems; i++) {
       double value = this->values[i];
       if (isPositive(value) && this->total > 0.0) {
          value = MINIMUM(value, this->total);
-         blockSizes[i] = ceil((value / this->total) * w);
+         actualBarWidth = ((value / this->total) * w);
+         blockSizes[i] = ceil(actualBarWidth);
       } else {
          blockSizes[i] = 0;
       }
       int nextOffset = offset + blockSizes[i];
       // (Control against invalid values)
       nextOffset = CLAMP(nextOffset, 0, w);
+
+      const Settings* settings = this->host->settings;
+      static const wchar_t* bars[8] = {
+         L" ||||||||",
+         L" ########",
+         L"⠀⡀⡄⡆⡇⣇⣧⣷⣿",
+         L" ░░▒▒▓▓██",
+         L" ▏▎▍▌▋▊▉█",
+         L" ▁▂▃▄▅▆▇█",
+         L" ▌▌▌▌████",
+         L" ▔🮂🮃▀🮄🮅🮆█"
+      };
+
       for (int j = offset; j < nextOffset; j++)
          if (RichString_getCharVal(bar, startPos + j) == ' ') {
-            if (CRT_colorScheme == COLORSCHEME_MONOCHROME) {
+            if (settings->barType) {
+               RichString_setChar(&bar, startPos + j, bars[settings->barType][7]);
+            } else if (CRT_colorScheme == COLORSCHEME_MONOCHROME) {
                assert(i < strlen(BarMeterMode_characters));
                RichString_setChar(&bar, startPos + j, BarMeterMode_characters[i]);
             } else {
                RichString_setChar(&bar, startPos + j, '|');
             }
          }
+
+      int barsLen=wcslen(bars[settings->barType]);
+      int subPixel = floor(actualBarWidth*(barsLen-1));
+      RichString_setChar(&bar, startPos + nextOffset - 1, bars[settings->barType][subPixel%(barsLen-1)]);
```

Comment:
Code style:

```suggestion
      const wchar_t *barChars = &bars[settings->barType][1];
      int barsLen = wcslen(barChars);
      int subPixel = floor(actualBarWidth * barsLen);
      RichString_setChar(&bar, startPos + nextOffset - 1, barChars[subPixel % barsLen]);
```

Storing the pointer to `bars[settings->barType][1]` simplifies the offset calculations below as the space does not need to be skipped. Empty cells still should use `bars[settings->barType][0]` and full ones `bars[settings->barType][barsLen]`.

Looking at the code, the floor at that position does not look right. Please re-check.